tox.ini

```diff
@@ -1,16 +1,43 @@
 [tox]
 minversion = 1.6
-envlist = coala
-skipsdist = true
+envlist =
+  pre-commit
+skipsdist = True
 
-[testenv:coala]
+[testenv:jjb]
 basepython = python3
 deps =
-    coala==0.11
-    coala-bears==0.11
-    nodeenv~=1.3.0
+    jenkins-job-builder==3.5.0
 commands =
-    nodeenv -p
-    npm install --global write-good
-    python3 -m nltk.downloader punkt maxent_treebank_pos_tagger averaged_perceptron_tagger
-    coala --non-interactive
+    jenkins-jobs -l DEBUG test --recursive -o {toxinidir}/archives/job-configs {toxinidir}/jjb:{toxinidir}/.jjb-test
+
+[testenv:jjb-version]
+basepython = python3
+commands = python {toxinidir}/check_jjb_version.py
+
+[testenv:pre-commit]
+basepython = python3
+allowlist_externals =
+    /bin/sh
+deps =
+    pre-commit
+passenv = HOME
+commands =
+    pre-commit run --all-files --show-diff-on-failure
+    /bin/sh -c 'if ! git config --get user.name > /dev/null; then \
+        git config --global --add user.name "CI"; \
+        touch .git/REMOVE_USERNAME; fi'
+    /bin/sh -c 'if ! git config --get user.email > /dev/null; then \
+        git config --global --add user.email "ci@example.org"; \
+        touch .git/REMOVE_USEREMAIL; fi'
+    /bin/sh -c "if [ -f .git/COMMIT_EDITMSG ]; then \
+        cp .git/COMMIT_EDITMSG .git/COMMIT_MSGTOX; else \
+        git log HEAD -n1 | tail -n +5 | cut -c 5- > .git/COMMIT_MSGTOX; fi"
```

Comment:
This line would be cleaner with what is here: https://github.com/lfit/releng-common-packer/blob/master/tox.ini#L39

Reply:
Updated